File: src/main/java/com/eh2/noframe/presentation/mapper/BookMapper.java
package com.eh2.noframe.presentation.mapper;

import com.eh2.noframe.domain.BookEntity;
import com.eh2.noframe.presentation.dto.book.BookResponseDTO;
import com.eh2.noframe.presentation.dto.book.CreateBookRequestDTO;

public class BookMapper {

	public BookEntity toBookEntity(CreateBookRequestDTO createBookRequestDTO) {
		return new BookEntity(createBookRequestDTO.title());
	}

	public BookResponseDTO toBookResponseDTO(BookEntity book) {
		return new BookResponseDTO(book.getId(), book.getTitle());
	}
}

File: src/main/kotlin/com/eh2/noframe/presentation/mapper/BookMapper.kt
package com.eh2.noframe.presentation.mapper

import com.eh2.noframe.domain.BookEntity
import com.eh2.noframe.presentation.dto.book.BookResponseDTO
import com.eh2.noframe.presentation.dto.book.CreateBookRequestDTO

class BookMapper {
    fun toBookEntity(createBookRequestDTO: CreateBookRequestDTO): BookEntity {
        return BookEntity(createBookRequestDTO.title)
    }

    fun toBookResponseDTO(book: BookEntity): BookResponseDTO {
        return BookResponseDTO(book.id, book.title)
    }
}
